motor_control/throttle/Button.h

```diff
@@ -0,0 +1,22 @@
+#ifndef BUTTON_THROTTLE
+#define BUTTON_THROTTLE
+
+#include "base_throttle.h"
+
+class Button: public Throttle{
+    public:
+    Button(int ft);
+    virtual int readThrottle();
+    int mappingFunction(int f, int s);
+    private:
+    int fullThrottle;//the amount of time the button must be pressed to bring the motor to full capacity
+    bool isPressed;
+    int pressStart;
```

Comment:
Private data members are good because they help you save state between iterations, so this is good stuff. But, let's stick to a common style of `m_variable_name`, as opposed to `variableName`, which is more common in Python variable naming.